ignite/pkg/cosmosclient/cosmosclient.go

```diff
@@ -390,6 +396,85 @@ func (c Client) BroadcastTxWithProvision(accountName string, msgs ...sdktypes.Ms
 	}, nil
 }
 
+// GetBlockTXs returns the transactions in a block.
+// The list of transactions can be empty if there are no transactions
+// in the block at the moment this method is called.
+func (c Client) GetBlockTXs(ctx context.Context, height int64) (txs []TX, err error) {
+	r, err := c.RPC.Block(ctx, &height)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch block %d: %w", height, err)
+	}
+
+	params := url.Values{}
+	params.Set(searchHeight, strconv.FormatInt(height, 10))
+	query := params.Encode()
+
+	// TODO: improve to fetch pages in parallel (requires fetching page 1 to calculate n. of pages)
+	page := 1
+	perPage := defaultTXsPerPage
+	blockTime := r.Block.Time
+	for {
+		res, err := c.RPC.TxSearch(ctx, query, false, &page, &perPage, orderAsc)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, tx := range res.Txs {
+			txs = append(txs, TX{
+				BlockTime: blockTime,
+				Raw:       tx,
+			})
+		}
+
+		// Stop when the last page is fetched
+		if res.TotalCount <= (page * perPage) {
+			break
+		}
+
+		page++
+	}
+
+	return txs, nil
+}
+
+// CollectTXs collects transactions from multiple consecutive blocks.
+// Transactions from a single block are send to the channel only if all transactions
+// from that block are collected successfully.
+// Blocks are traversed sequentially starting from a height until the latest block height
+// available at the moment this method is called.
+// The channel might contain the transactions collected successfully up until that point
+// when an error is returned.
+func (c Client) CollectTXs(ctx context.Context, fromHeight int64, tc chan<- []TX) error {
+	defer close(tc)
+
+	status, err := c.Status(ctx)
```

Comment:
Shouldn't we continuously check status to always get the latest height, or we plan to recall CollectTXs once it returns a non-nil error?

Reply:
I was planning to leave the responsibility on how to handle the recurrent calls and height "tracking" to the caller to keep `cosmosclient.Client` simple.

So the caller or "service" would call with height "0" the very first time, because there are no transactions indexed at that point. After that it will perform recurrent calls to `CollectTXs()` using the height of the last block that was indexed. This height can be read using the `Adapter.GetLatestHeight()` method. The frequency of the calls would also be configured in the caller.

A call to `CollectTXs()` can fail with an error in which case the caller would be able to properly deal with the error by logging it and then use any other measures needed to continue, like retrying a couple of times or whatever is deemed necessary.